command/kv_delete.go

```diff
@@ -0,0 +1,140 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+var _ cli.Command = (*KVDeleteCommand)(nil)
+var _ cli.CommandAutocomplete = (*KVDeleteCommand)(nil)
+
+type KVDeleteCommand struct {
+	*BaseCommand
+
+	flagVersions []string
+}
+
+func (c *KVDeleteCommand) Synopsis() string {
+	return "Deletes versions in the KV store"
+}
+
+func (c *KVDeleteCommand) Help() string {
+	helpText := `
+Usage: vault kv delete [options] PATH
+
+  Deletes the data for the provided version and path in the key-value store. The
+  versioned data will not be fully removed, but marked as deleted and  will no
+  longer be returned in normal get requests.
+
+  To delete the lastest version of the key "foo": 
+
+      $ vault kv delete secret/foo
+
+  To delete version 3 of key foo:
+
+      $ vault kv delete -versions=3 secret/foo
+
+  Additional flags and more advanced use cases are detailed below.
```

Comment:
Should we also reference here that the metadata command can delete all versions?  That not obvious IMHO, and I'd be searching for a variant of `delete` to do that.